### PR TITLE
Upgrade python to 3.9

### PR DESCRIPTION
**Pull request by marcospri**
~~Just curious now, will this build?~~

Python upgrade to 3.9

Python 3.9 come out in 2020 so libraries had time to adapt to any changes.

There's nothing ground braking on the change log:  https://docs.python.org/3/whatsnew/3.9.html

but I made a few non-required changes to take advantage of the new features:

- [x] Use https://peps.python.org/pep-0615/ instead the backport
- [x] Replace some typing import for https://peps.python.org/pep-0585/

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -17,7 +17,6 @@ on:
       - 'requirements/dev.txt'
       - '.docker.env'
       - '**/.gitignore'
-      - '.python-version'
```
Comment (marcospri) [inline .github/workflows/ci.yml:20]: This doesn't happen often enough to ignore it.

**Comment by robertknight** [comment]
What is the thinking behind going to Python 3.9 rather than straight to 3.11?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (seanh) [reply .github/workflows/ci.yml:20]: :+1: Actually I think we _do_ want to run CI on a PR that just changes `.python-version`, so I don't think this should have been here.

`ci.yml` comes from the cookiecutter so you'll have to backport this change to the cookiecutters repo

File: .cookiecutter/cookiecutter.json [hunk 2]
```diff
@@ -18,7 +18,7 @@
     "package_name": "lms",
     "slug": "lms",
     "short_description": "An app for integrating Hypothesis with Learning Management Systems (LMS's).",
-    "python_version": "3.8.12",
+    "python_version": "3.9.17",
```
Comment (seanh) [inline .cookiecutter/cookiecutter.json:21]: :+1:

**Review by seanh — APPROVED** [review]

**Comment by marcospri** [comment]
Closing this for now, see 

https://hypothes-is.slack.com/archives/C1MA4E9B9/p1692631078042749

**Pull request closed without merging**